pkg/diag/validator/pod.go

```diff
@@ -120,37 +120,80 @@ func (p *PodValidator) getPodStatus(pod *v1.Pod) *podStatus {
 
 func getPodStatus(pod *v1.Pod) (proto.StatusCode, []string, error) {
 	// See https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions
-	for _, c := range pod.Status.Conditions {
-		if c.Type == v1.PodScheduled {
-			switch c.Status {
-			case v1.ConditionFalse:
-				logrus.Debugf("Pod %q not scheduled: checking tolerations", pod.Name)
-				sc, err := getUntoleratedTaints(c.Reason, c.Message)
-				return sc, nil, err
-			case v1.ConditionTrue:
-				logrus.Debugf("Pod %q scheduled: checking container statuses", pod.Name)
-				// TODO(dgageot): Add EphemeralContainerStatuses
-				cs := append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...)
-				// See https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-states
-				statusCode, logs, err := getContainerStatus(pod, cs)
-				if statusCode == proto.StatusCode_STATUSCHECK_POD_INITIALIZING {
-					// Determine if an init container is still running and fetch the init logs.
-					for _, c := range pod.Status.InitContainerStatuses {
-						if c.State.Waiting != nil {
-							return statusCode, []string{}, fmt.Errorf("waiting for init container %s to start", c.Name)
-						} else if c.State.Running != nil {
-							return statusCode, getPodLogs(pod, c.Name), fmt.Errorf("waiting for init container %s to complete", c.Name)
-						}
-					}
+	if isPodReady(pod) {
```

Comment:
It would also be useful to explain in prose the different conditions, especially odd cases like `the pod can be ready, but a container could be not ready because it terminated.`

Reply:
done